stdlib/public/Cxx/CxxSpan.swift

```diff
@@ -36,6 +47,28 @@ extension CxxSpan {
                   "UnsafeMutableBufferPointer should not point to nil")
     self.init(unsafeMutableBufferPointer.baseAddress!, Size(unsafeMutableBufferPointer.count))
   }
+
+  @available(SwiftStdlib 6.1, *)
+  @inlinable
+  @unsafe
+  public init(_ span: Span<Element>) {
+    let (p, c) = unsafeBitCast(span, to: (UnsafeRawPointer?, Int).self)
+    precondition(p != nil, "Span should not point to nil")
+    let binding = p!.bindMemory(to: Element.self, capacity: c)
+    self.init(binding, Size(c))
+  }
+}
+
+@available(SwiftStdlib 6.1, *)
+extension Span {
+  @_alwaysEmitIntoClient
+  @lifetime(immortal)
+  @unsafe
+  public init<T: CxxSpan<Element>>(
+    _unsafeCxxSpan span: T,
```

Comment:
I think an implicit convention that we've followed with these initializers so far has been to call the parameter `fromCxx:`, but given that this is underscored, I guess we can bikeshed the naming later.

Reply:
Hmm. I think in this particular case we really want `unsafe` to be part of the name because the compiler does not reason about the lifetimes here. It could be `_unsafeFromCxx`.